common/lib/src/realm_types.dart

```diff
@@ -79,6 +79,36 @@ class RealmStateError extends StateError implements RealmError {
   RealmStateError(String message) : super(message);
 }
 
+/// Thrown or reporeted if an error occurs during synchronization
+/// {@category Sync}
+class SyncError extends RealmError {
```

Comment:
Could we rename SyncError  to SyncErrorCode? upload_completion and  download_completion are working with realm_sync_error_code. We need to have another class SyncError for sync_error_handler argument.

Reply:
It's quite unfortunate we're using different error types in Core. `SyncErrorCode` doesn't really make sense as a name in dart as I'd imagine most developers will expect an error to be suffixed with `Error`. We should probably collapse both types to a single one in dart as `realm_sync_error_t` doesn't really differ that much.

Reply:
OK. If I remove the properties that you suggested from SyncError, then they will be the same.

Reply:
Actually even though I remove most of the redundant properties, both errors are still different:
```dart
class SyncErrorCode extends RealmError {
  /// The code of the error
  final int code; // TODO: this should be an enum

  /// The category of the error
  final SyncErrorCategory category;

  SyncErrorCode(String message, this.category, this.code) : super(message);
}

class SyncError extends RealmError {
  final SyncErrorCode errorCode;
  final bool isFatal;

  SyncError(
    this.errorCode,
    String message,
    this.isFatal
    ) : super(message);
}
```